File: app/src/main/java/com/bopr/android/smailer/ui/BaseSettingsListener.java
package com.bopr.android.smailer.ui;

import android.app.backup.BackupManager;
import android.content.Context;
import android.content.SharedPreferences;

public class BaseSettingsListener implements SharedPreferences.OnSharedPreferenceChangeListener {

    private BackupManager backupManager;

    BaseSettingsListener(Context context) {
        backupManager = new BackupManager(context);
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        backupManager.dataChanged();
    }
}

File: app/src/main/kotlin/com/bopr/android/smailer/ui/BaseSettingsListener.kt
package com.bopr.android.smailer.ui

import android.app.backup.BackupManager
import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.OnSharedPreferenceChangeListener

open class BaseSettingsListener internal constructor(context: Context) : OnSharedPreferenceChangeListener {

    private val backupManager: BackupManager = BackupManager(context)

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
        backupManager.dataChanged()
    }

}